pinot-segment-spi/src/main/java/org/apache/pinot/segment/spi/index/metadata/ColumnMetadataImpl.java

```diff
@@ -355,7 +415,9 @@ public static class Builder {
     private PartitionFunction _partitionFunction;
     private Set<Integer> _partitions;
     private boolean _autoGenerated;
-    private Map<IndexType<?, ?, ?>, Long> _indexSizeMap = new HashMap<>();
+
+    // use non-default size to save space for most columns
+    LongArrayList _indexSizeList = new LongArrayList(2);
```

Comment:
This should be updated in the builder

Reply:
What do you mean? Should it be done in build() method or by explicit call outside builder ?

Reply:
With builder pattern, we want to initialize everything in the builder, and not modify the class once `build()` is invoked. Right now the update happens in the main class, instead of the builder

Reply:
Collection is in the builder already. By update did you mean adding to index size map ? Code responsible for that  puts data into ColumnMetadataImpl ( way after instance is built by builder)  in master already. Should I refactor SegmentMetadataImpl then ? I was trying to minimize changes so far.